### DISCUSSION TITLE
Cant't enable Chromium/Chrome

### DISCUSSION THREAD

**Discussion by DarioGui** [YetiForceCompany/YetiForceCRM#17106] [How to]
Hello everyone!
I'm definitely doing something wrong but I can't get Chromium/Chrome activated.
I purchased the add-on extension (it's been months now but I still left the payment even though I can't use it.. at least I contribute something) but I can't seem to enable the pdf engine.
I changed the path in the pdf.php file and pointed to the path they provided me from the hosting (/usr/bin/google-chrome).
I replaced the last line with
"public static $chromiumBrowserOptions = ['noSandbox' => true, 'noProxyServer' => true, 'keepAlive' => true];"
But when I go to PDF I only ever see the standard pdf engine.
From the hosting they told me that the engine is installed but maybe I need to install some additional library? But in the yetiforce guide it doesn't say anything.
Sorry to ask but it is too complicated for me here :)

**Comment by rskrzypczak** [comment]
https://doc.yetiforce.com/administrator-guides/system-tools/pdf/#configuring-the-pdf-generator-path

**Reply by DarioGui** [reply]
Thx @rskrzypczak!
I used those instructions to edit the pdf.php file.
I entered the path location (as the hosting service people told me) and changed the line "public static $chromiumBrowserOptions" because it was different from the one contained in my pdf.php file.
However, to be safe, this is my file:

`<?php

/**
 * Configuration file.
 * This file is auto-generated.
 *
 * @package Config
 *
 * @copyright YetiForce S.A.
 * @license   YetiForce Public License 5.0 (licenses/LicenseEN.txt or yetiforce.com)
 */

namespace Config\Components;

/**
 * Configuration file: Config\Components\Pdf.
 */
class Pdf
{
	/**
	 * The name or path of the chrome/chromium engine.
	 *
	 * @see https://www.chromium.org/getting-involved/download-chromium
	 *
	 * @var string
	 */
	public static $chromiumBinaryPath = '/usr/bin/google-chrome';

	/**
	 * Chromium browser options available for the browser factory.
	 *
	 * @see https://github.com/chrome-php/chrome#available-options
	 *
	 * @var array
	 */
	public static $chromiumBrowserOptions = ['noSandbox' => true, 'noProxyServer' => true, 'keepAlive' => true];
}`

**Comment by mariuszkrzaczkowski** [comment]
run this from the CLI on the OS of the same account that owns the CRM files
`/usr/bin/google-chrome --version`

**Comment by DarioGui** [comment]
the answer is: Google Chrome 112.0.5615.49

**Reply by mariuszkrzaczkowski** [reply]
check the logs

**Reply by DarioGui** [reply]
Sorry for the delay but I am really ignorant about these things :)
searching the log I found that the proc_open function is not available on my installation.
I have written to my provider now and am waiting for them to enable it.
Could this just be the problem?

[warning][[App\Pdf\Drivers\Chromium]] - Symfony\Component\Process\Exception\LogicException: The Process class relies on proc_open, which is not available on your PHP installation. in [...] public_html/vendor/symfony/process/Process.php:146
public_html/vendor/chrome-php/chrome/src/Browser/BrowserProcess.php(127): Symfony\Component\Process\Process->__construct()
public_html/vendor/chrome-php/chrome/src/BrowserFactory.php(81): HeadlessChromium\Browser\BrowserProcess->start()
public_html/app/Pdf/Drivers/Chromium.php(118): HeadlessChromium\BrowserFactory->createBrowser()
public_html/app/Pdf/Pdf.php(64): App\Pdf\Drivers\Chromium::isActive()

**Reply by mariuszkrzaczkowski** [reply]
> Sorry for the delay but I am really ignorant about these things :) searching the log I found that the proc_open function is not available on my installation. I have written to my provider now and am waiting for them to enable it. Could this just be the problem?

it's definitely a problem but will it be the last one to see how they fix it

**Comment by DarioGui** [comment]
Hello everyone!
After many messages with support we solved that problem and several others that happened in the meantime.
Now, however, it turns out to be yet another error. In the logs I see:

[warning][[App\Pdf\Drivers\Chromium]] - Wrench\Exception\HandshakeException: No accept header receieved on handshake response in 

My version of Chrome is the one written above (112.0.5615.49).
Do I understand correctly that I need to have Chrome downgraded?
Or do you have another better suggestion?

thank you very much for all the help

**Comment by DarioGui** [comment]
Sorry
but by mistake I wrote in two posts.
Here and in https://github.com/YetiForceCompany/YetiForceCRM/issues/17039
